stubs/toml/toml/decoder.pyi

```diff
@@ -0,0 +1,79 @@
+import sys
+from _typeshed import SupportsRead
+from typing import Any, Callable, Generic, MutableMapping, Pattern, Protocol, Text, TypeVar, overload
+
+_MutableMappingT = TypeVar("_MutableMappingT", bound=MutableMapping[str, Any])
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT = TypeVar("_VT")
+
+if sys.version_info >= (3, 0):
+    from pathlib import PurePath
+
+    FNFError = FileNotFoundError
+    _PathLike = str | bytes | PurePath
+else:
+    FNFError = IOError
+    _PathLike = str | bytes
+
+class _SupportsGetSetItem(Protocol[_KT_contra, _VT]):
+    def __getitem__(self, __k: _KT_contra) -> _VT: ...
+    def __setitem__(self, __k: _KT_contra, __v: _VT) -> None: ...
+
+TIME_RE: Pattern[str]
+
+class TomlDecodeError(ValueError):
+    msg: str
+    doc: str
+    pos: int
+    lineno: int
+    colno: int
+    def __init__(self, msg: str, doc: str, pos: int) -> None: ...
+
+class CommentValue(_SupportsGetSetItem[_KT_contra, _VT]):
+    val: _SupportsGetSetItem[_KT_contra, _VT]
+    comment: str
+    def __init__(
+        self, val: _SupportsGetSetItem[_KT_contra, _VT], comment: str, beginline: bool, _dict: type[_MutableMappingT]
+    ) -> None: ...
+    def __getitem__(self, key: _KT_contra) -> _VT: ...
+    def __setitem__(self, key: _KT_contra, value: _VT) -> None: ...
+    def dump(self, dump_value_func: Callable[[Any], str]) -> str: ...
+
+@overload
+def load(
+    f: _PathLike | list[_PathLike] | SupportsRead[Text],
+    _dict: type[dict[str, Any]] = ...,
+    decoder: TomlDecoder[dict[str, Any]] | None = ...,
+) -> dict[str, Any]: ...
+@overload
+def load(
+    f: _PathLike | list[_PathLike] | SupportsRead[Text],
+    _dict: type[_MutableMappingT],
+    decoder: TomlDecoder[_MutableMappingT] | None = ...,
+) -> _MutableMappingT: ...
+@overload
+def loads(s: Text, _dict: type[_MutableMappingT] = ..., decoder: TomlDecoder[dict[str, Any]] | None = ...) -> dict[str, Any]: ...
+@overload
+def loads(s: Text, _dict: type[_MutableMappingT], decoder: TomlDecoder[_MutableMappingT] | None = ...) -> _MutableMappingT: ...
+
+class InlineTableDict: ...
+
+class TomlDecoder(Generic[_MutableMappingT]):
+    _dict: type[_MutableMappingT]
+    def __init__(self, _dict: type[_MutableMappingT] = ...) -> None: ...
```

Comment:
I think this needs overloads to make `_dict` be `dict[str, Any]` by default, as `dict[str, Any]` isn't currently even mentioned anywhere.

```suggestion
    @overload
    def __init__(self: TomlDecoder[dict[str, Any]], _dict: type[dict[str, Any]] = ...) -> None: ...
    @overload
    def __init__(self, _dict: type[_MutableMappingT]) -> None: ...
```